CMakeLists.txt

```diff
@@ -0,0 +1,294 @@
+    CONFIGURE_DEPENDS
+    LIST_DIRECTORIES false
+    src/*.cpp
+    src/*.c
+)
+
+# Headers
+file(GLOB_RECURSE bb_headers
+    RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}
+    CONFIGURE_DEPENDS
+    LIST_DIRECTORIES false
+    src/*.h
+    src/*.hpp
+    src/*.inl
+)
+
+# Configure dependent on config/platform/architecture
+list(FILTER bb_sources EXCLUDE REGEX "src/main\.cpp")
+list(FILTER bb_sources EXCLUDE REGEX "src/(test|platform)/.+")
+list(FILTER bb_sources EXCLUDE REGEX "src/b3/blake3_(avx|sse).+")
+
+# Architecture
+if(${CMAKE_HOST_SYSTEM_PROCESSOR} MATCHES "x86_64" OR ${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "AMD64")
+    
+    list(APPEND bb_sources
+        src/b3/blake3_avx2.c
+        src/b3/blake3_avx512.c
+        src/b3/blake3_sse41.c
+    )
+
+    if(NOT MSVC)
+        list(APPEND bb_sources
+            src/b3/blake3_avx2_x86-64_unix.S
+            src/b3/blake3_avx512_x86-64_unix.S
+            src/b3/blake3_sse41_x86-64_unix.S
+        )
+    else()
+        # Disable blake3 convertion loss of data warnings 
+        set_source_files_properties( 
+            src/b3/blake3_avx2.c
+            src/b3/blake3_avx512.c
+            src/b3/blake3_sse41.c
+            PROPERTIES COMPILE_FLAGS
+            /wd4244
+        )
+    endif()
+elseif(${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "arm64" OR ${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "aarch64")
+else()
+    message( FATAL_ERROR "Unsupported architecture '${CMAKE_HOST_SYSTEM_PROCESSOR}'" )
+endif()
+
+# OS
+if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+
+    file(GLOB_RECURSE src_linux RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} 
+        CONFIGURE_DEPENDS LIST_DIRECTORIES false
+        src/platform/unix/*.cpp
+        src/platform/linux/*.cpp
+    )
+    list(APPEND bb_sources ${src_linux})
+
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+    
+    file(GLOB_RECURSE src_win RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} 
+        CONFIGURE_DEPENDS LIST_DIRECTORIES false
+        src/platform/win32/*.cpp
+    )
+    list(APPEND bb_sources ${src_win})
+
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+
+    file(GLOB_RECURSE src_mac RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} 
+        CONFIGURE_DEPENDS LIST_DIRECTORIES false
+        src/platform/unix/*.cpp
+        src/platform/macos/*.cpp
+    )
+    list(APPEND bb_sources ${src_mac})
+
+else()
+    message( FATAL_ERROR "Unsupported operating system '${CMAKE_SYSTEM_NAME}'" )
+endif()
+
+# Dev config?
+file(GLOB_RECURSE src_dev RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} 
+    CONFIGURE_DEPENDS LIST_DIRECTORIES false
+    src/test/*.cpp
+)
+
+file(GLOB_RECURSE headers_dev RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} 
+    CONFIGURE_DEPENDS LIST_DIRECTORIES false
+    src/test/*.h
+)
+
+
+# Exe
+find_package(Threads REQUIRED)
+
+set(platform_libs)
+
+if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+    set(platform_libs numa)
+endif()
+
+set(bb_include_dirs 
+    ${INCLUDE_DIRECTORIES}
+    ${CMAKE_CURRENT_SOURCE_DIR}/src
+    ${bls_SOURCE_DIR}/include
+    ${relic_SOURCE_DIR}/include
```

Comment:
You'd only need `${CMAKE_CURRENT_SOURCE_DIR}/src` here, as you get the `bls` and `relic` ones as part of usage requirements.